File: lessons/gen_scripts/php/scripts/e30.branch.php
#!/usr/bin/env php
<?php
// PHP heredoc strips exactly the trailing newline right before the
//  closing marker, so this ends in "...Make your selection: " with no
//  linebreak before the answer is typed - no extra trim needed.
$menu = <<<EOT
Select an item from the menu.

  1 - Coffee
  2 - Espresso
  3 - Latte
  4 - Machiato
  5 - Capucino
  6 - Mocha
  7 - Tea

Make your selection: 
EOT;
echo $menu;
$selection = trim(fgets(STDIN));

if ($selection == 1) {
  echo "You selected a Coffee\n";
} elseif ($selection == 2) {
  echo "You selected an Espresso\n";
} elseif ($selection == 3) {
  echo "You selected a Latte\n";
} elseif ($selection == 4) {
  echo "You selected a Machiato\n";
} elseif ($selection == 5) {
  echo "You selected a Capucino\n";
} elseif ($selection == 6) {
  echo "You selected a Mocha\n";
} elseif ($selection == 7) {
  echo "You selected a Tea\n";
} else {
  echo "You have not entered a valid selection\n";
}
?>

File: lessons/gen_scripts/ruby/scripts/e30.branch.rb
#!/usr/bin/env ruby
# .chomp strips the heredoc's own trailing newline - no linebreak before
#  the answer is typed
menu = <<~END.chomp
  Select an item from the menu.

    1 - Coffee
    2 - Espresso
    3 - Latte
    4 - Machiato
    5 - Capucino
    6 - Mocha
    7 - Tea

  Make your selection: 
END
print menu
$stdout.flush
selection = gets.chomp.to_i

if selection == 1
  puts "You selected a Coffee"
elsif selection == 2
  puts "You selected an Espresso"
elsif selection == 3
  puts "You selected a Latte"
elsif selection == 4
  puts "You selected a Machiato"
elsif selection == 5
  puts "You selected a Capucino"
elsif selection == 6
  puts "You selected a Mocha"
elsif selection == 7
  puts "You selected a Tea"
else
  puts "You have not entered a valid selection"
end
